airbyte-integrations/connectors/source-stripe/unit_tests/integration/test_events.py

```diff
@@ -0,0 +1,350 @@
+
+
+def _source(catalog: ConfiguredAirbyteCatalog) -> SourceStripe:
+    return SourceStripe(catalog)
+
+
+def _create_builders() -> Tuple[RecordBuilder, HttpResponseBuilder]:
+    return create_builders_from_resource(
+        find_template("events", __file__),
+        FieldPath("data"),
+        record_id_path=FieldPath("id"),
+        record_cursor_path=FieldPath("created"),
+        pagination_strategy=StripePaginationStrategy(),
+    )
+
+
+def _a_record() -> RecordBuilder:
+    return _create_builders()[0]
+
+
+def _a_response() -> HttpResponseBuilder:
+    return _create_builders()[1]
+
+
+def _read(
+    config: ConfigBuilder,
+    sync_mode: SyncMode,
+    state: Optional[Dict[str, Any]] = None,
+    expecting_exception: bool = False
+) -> EntrypointOutput:
+    catalog = _catalog(sync_mode)
+    return read(_source(catalog), config.build(), catalog, state, expecting_exception)
+
+
+@freezegun.freeze_time(_NOW.isoformat())
+class FullRefreshTest(TestCase):
+
+    @HttpMocker()
+    def test_given_one_page_when_read_then_return_record(self, http_mocker: HttpMocker) -> None:
+        http_mocker.get(
+            HttpRequest(
+                url="https://api.stripe.com/v1/events",
+                query_params={"created[gte]": str(int(_A_START_DATE.timestamp())), "created[lte]": str(int(_NOW.timestamp())), "limit": 100},
+                headers=_AUTHENTICATION_HEADERS
+            ),
+            _a_response().with_record(_a_record()).build()
+        )
+        output = self._read(_config().with_start_date(_A_START_DATE))
+        assert len(output.records) == 1
+
+    @HttpMocker()
+    def test_given_many_pages_when_read_then_return_records(self, http_mocker: HttpMocker) -> None:
+        http_mocker.get(
+            HttpRequest(
+                url="https://api.stripe.com/v1/events",
+                query_params={"created[gte]": str(int(_A_START_DATE.timestamp())), "created[lte]": str(int(_NOW.timestamp())), "limit": 100},
+                headers=_AUTHENTICATION_HEADERS,
+            ),
+            _a_response().with_pagination().with_record(_a_record().with_id("last_record_id_from_first_page")).build()
+        )
+        http_mocker.get(
+            HttpRequest(
+                url="https://api.stripe.com/v1/events",
+                query_params={
+                    "starting_after": "last_record_id_from_first_page",
+                    "created[gte]": str(int(_A_START_DATE.timestamp())),
+                    "created[lte]": str(int(_NOW.timestamp())),
+                    "limit": 100,
+                },
+                headers=_AUTHENTICATION_HEADERS,
+            ),
+            _a_response().with_record(_a_record()).build()
+        )
+        output = self._read(_config().with_start_date(_A_START_DATE))
+        assert len(output.records) == 2
+
+    @HttpMocker()
+    def test_given_start_date_before_30_days_stripe_limit_when_read_then_request_more_than_30_days(self, http_mocker: HttpMocker) -> None:
```

Comment:
It is a special case because Stripe will only ever return 30 days worth of data on events. The [doc](https://stripe.com/docs/api/events) mentions: `We only guarantee access to events through the [Retrieve Event API](https://stripe.com/docs/api/events#retrieve_event) for 30 days.` Hence, this feels noteworthy